### DISCUSSION TITLE
Create approle with vault_write module

### DISCUSSION THREAD

**Discussion by dil-kpogany** [ansible-collections/community.hashi_vault#278] [Q&A] [answered]
Hi,

Is there a way to use the` vault_write` module for `approle` creation?

Thanks

**Comment by briantist** [comment, marked as answer]
Hi @dil-kpogany , welcome!

Do you mean initial creation of the approle? I haven't tried that myself but I think it would be possible, yes.

`vault_write` is essentially an HTTP `POST` to the given path, so as long as the APIs needed use `POST` you should be able to do it.

The relevant docs for that are in Vault's HTTP API docs:
https://www.vaultproject.io/api-docs/auth/approle#create-update-approle

So based on that, I think you can `vault_write` to the path `auth/approle/role/:role_name` to do the creation. The data would be a dict of the parameters listed on that page.

Let me know if that works for you!

---

Also, I expect that eventually we'll have a dedicated plugin/module for that, but it will require someone to contribute it. I don't have enough time to create all of the ones I would like to have, so I generally create them when I need them, or when I happen to have free time (increasingly rare), prioritized roughly by impact and usefulness.

If you're interested in contributing, we have a [Contributor Guide](https://docs.ansible.com/ansible/latest/collections/community/hashi_vault/docsite/contributor_guide.html) to help get started.

**Reply by dil-kpogany** [reply]
Thanks a lot, I'll try that.
